### PR TITLE
Add to_linear_operator to lindblad (so can use with scikit.integrate) and improve nk.exact.steadystate

### PR DESCRIPTION
**Pull request by PhilipVinc**
also add tests

### CODE REVIEW

File: netket/exact.py [hunk 1]
```diff
@@ -194,46 +200,20 @@ def steady_state(lindblad, sparse=False, method="ed", rho0=None, **kwargs):
             w, v = eigsh(ldagl, which="SM", k=2)
 
         print("Minimum eigenvalue is: ", w[0])
-        rho = matrix(v[:, 0].reshape((M, M)))
+        rho = array(v[:, 0].reshape((M, M)))
```
Comment (femtobit) [inline netket/exact.py:203]: After rebasing on `master`, `v[:, 0].reshape((M, M))` is always an array.
```suggestion
        rho = v[:, 0].reshape((M, M))
```

File: netket/exact.py [hunk 2]
```diff
@@ -143,7 +143,7 @@ def full_ed(operator, compute_eigenvectors=False):
         return EdResult(w, None)
 
 
-def steady_state(lindblad, sparse=False, method="ed", rho0=None, **kwargs):
+def steady_state(lindblad, sparse=None, method="ed", rho0=None, **kwargs):
```
Comment (femtobit) [inline netket/exact.py:146]: Maybe make the options here keyword-only as well, while we're at it?
```suggestion
def steady_state(lindblad, *, sparse=None, method="ed", rho0=None, **kwargs):
```

File: netket/operator/_local_liouvillian.py [hunk 3]
```diff
@@ -299,3 +301,78 @@ def _get_conn_flattened_kernel(
             sections[i] = off
 
         return _np.copy(xs[0:off, :]), _np.copy(mels[0:off])
+
+    def to_linear_operator(
+        self, sparse: bool = True, append_trace: bool = False
```
Comment (femtobit) [inline netket/operator/_local_liouvillian.py:306]: Maybe also:
```suggestion
        self, *, sparse: bool = True, append_trace: bool = False
```

**Comment by femtobit** [comment]
Looks good to me code-wise. What is the issue in the `test_exact_ss_ed` test?

**Comment by PhilipVinc** [comment]
it's a precision issue in the ED... let me fix the test

**Comment by PhilipVinc** [comment]
I have no idea why this stopped working all of a sudden.
Is it possible that they updated some dependency and now I'm doing something wrong..?

**Comment by PhilipVinc** [comment]
Indeed, seems like the sparse diagonalization works but not the dense

**Comment by PhilipVinc** [comment]
Bump.
Can someone approve (or comment?)

**Review by gcarleo — APPROVED** [review]
yes this looks good

**Pull request merged**